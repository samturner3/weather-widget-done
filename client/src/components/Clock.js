import React from 'react';
import moment from 'moment';
import 'moment-timezone';
import PropTypes from 'prop-types';

export default class TimeClock extends React.Component {
	constructor(props) {
		super(props);
		const UTCtime = moment().utc().clone();
		const localTime = UTCtime.add(this.props.UtcOffSet, 'm');
		this.state = {
			time: localTime.format('h:mm a'),
			day: localTime.format('dddd'),
			month: localTime.format('Do MMMM'),
			year: localTime.format('gggg')
		};
		this.displayTime = this.displayTime.bind(this);
	}

	componentDidMount() {
		// Arrow function allows you to use "this" in context of the Component
		this.interval = setInterval(() => {
			this.displayTime();
		}, 1000);
	};

	displayTime() {
		this.setState({
			time: moment().utc().add(this.props.UtcOffSet, 'm').clone().format('h:mm:ss a')
		});
	}
	render() {
		return (
			<div>
				<p><i>Local Time:</i></p>
				<p>{this.state.day}</p>
				<p>{this.state.month} {this.state.year} {this.state.time}</p>
			</div>
		);
	}
}

TimeClock.propTypes = {
	UtcOffSet: PropTypes.number.isRequired
}
