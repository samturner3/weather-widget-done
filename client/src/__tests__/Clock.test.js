import React from 'react';
import Clock from '../components/Clock';

function setup() {
	const props = {
		UtcOffSet: 60
	};

	return {
		wrapper: shallow(<Clock {...props} />)
	};
}

describe('<Clock />', () => {
	it('should render all elements', () => {
		const { wrapper } = setup();
		expect(wrapper).toMatchSnapshot();
	});
});
