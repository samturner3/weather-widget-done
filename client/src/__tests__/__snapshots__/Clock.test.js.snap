// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> should render all elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimeClock
    UtcOffSet={60}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          <i>
                    Local Time:
          </i>
</p>,
        <p>
          Friday
</p>,
        <p>
          24th August
           
          2018
           
          1:21 am
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i>
            Local Time:
</i>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Local Time:",
          },
          "ref": null,
          "rendered": "Local Time:",
          "type": "i",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Friday",
        },
        "ref": null,
        "rendered": "Friday",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "24th August",
            " ",
            "2018",
            " ",
            "1:21 am",
          ],
        },
        "ref": null,
        "rendered": Array [
          "24th August",
          " ",
          "2018",
          " ",
          "1:21 am",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <i>
                        Local Time:
            </i>
</p>,
          <p>
            Friday
</p>,
          <p>
            24th August
             
            2018
             
            1:21 am
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i>
              Local Time:
</i>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Local Time:",
            },
            "ref": null,
            "rendered": "Local Time:",
            "type": "i",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Friday",
          },
          "ref": null,
          "rendered": "Friday",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "24th August",
              " ",
              "2018",
              " ",
              "1:21 am",
            ],
          },
          "ref": null,
          "rendered": Array [
            "24th August",
            " ",
            "2018",
            " ",
            "1:21 am",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
